### DISCUSSION TITLE
Beginner question, correct way to test custom module?

### DISCUSSION THREAD

**Discussion by chvvkumar** [AllskyTeam/allsky-modules#201] [Q&A] [answered]
I am writing a module that will pull a JSON object from MQTT and add this to the overlay variables. The link to the module I am working on is:

https://github.com/chvvkumar/allsky-modules/tree/allsky_mqttsubscribe/allsky_mqttsubscribe

What is the correct way to test this on my AllSky instance? Primary concern is, if I clone my repo to the running pi and run install.sh, would it overwrite the modules already installed from the official allsky-modules repo  (or even uninstall ehm since I will only select my own module to install in my repo)?

Or is there a way to test the module from `~/allsky/config/myFiles/modules`? I read [the doc](http://allskypi5.lan/documentation/modules/modules.html#:~:text=Developing%20Custom%20Modules,-The%20module%20system) but it was not clear to me how the files in the path are handled. Do I just copy my script and requirements there?

**Comment by Alex-developer** [comment]
Best way is to copy the module file into /opt/allsky/modules. If it requires any dependencies then manually install them, don't forget to enable the venv first

Once its in there it will be available in the module manager.

Keep an eye on /var/log/lighttpd for any errors as whilst he modules are written in Python the metadata variable is read by php so if there are any issues with it you may get some errors.

**Reply by chvvkumar** [reply]
Thank you!

**Reply by chvvkumar** [reply]
I created a PR for my changes. I saw your message to another PR and created mine targeting the Dev branch. But I see there are commits ahead of me. The PR I created is:

https://github.com/AllskyTeam/allsky-modules/pull/203

Does this look correct?

**Reply by chvvkumar** [reply]
@Alex-developer Not sure if a closed discussion will trigger a notification but I had the above question about submitting a PR. Would it be better to open a new topic?

**Reply by Alex-developer** [reply, marked as answer]
> @Alex-developer Not sure if a closed discussion will trigger a notification but I had the above question about submitting a PR. Would it be better to open a new topic?

Yes it does trigger a notification :-)

I recreated the dev branch a few hours ago as it was a long way out of date. If you do a new PR against the dev branch it should be ok

Alex
